src/tests/universal-nrp-e2e/UniversalNRP.Tests.ps1

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# UniversalNRP.Tests.ps1
+param (
+    [Parameter(Mandatory)]
+    [string] $PolicyPackage
+)
+
+Describe 'Validate Universal NRP' {
+    # Perform monitoring - Get
+    Context "Get" {
+        BeforeAll {
+            $result = Get-GuestConfigurationPackageComplianceStatus -Path $PolicyPackage -Verbose
+        }
+
+        It 'Ensure the total resource instances count' {    
+            $result | Should -Not -BeNullOrEmpty
+            $result.resources | Should -HaveCount 20
+        }
```

Comment:
Can we look in results and pick up the code and reason and check those two individually?

Reply:
Added checks for code